nodejs/awsx/ec2/vpc.ts

```diff
@@ -429,9 +377,25 @@ export interface VpcSubnetArgs {
      *
      * The allowed mask size is between a 28 netmask and 16 netmask.  See
      * https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Subnets.html for more details.
+     *
+     * If this property is provided, [location] cannot be provided.
      */
     cidrMask?: number;
 
+    /**
+     * More precise information about the location of this subnet.  Can either be a simple CidrBlock
+     * (i.e. 10.0.0.0/24), or a richer object describing the CidrBlocks and Availability Zone for
+     * the subnet.
+     *
+     * If this property is provided, [cidrMask] cannot be provided.
+     *
+     * If only a CidrBlock is provided here, then the subnet will be placed in the first
+     * availability zone for the region.
+     *
+     * If this property is provided for one subnet, it must be provided for all subnets.
+     */
+    location?: CidrBlock | VpcSubnetLocation;
```

Comment:
this is the new way that clients can pass in specific information to control making the subnet.   It basically says "i'm going to control all of this.".